mws/resource_log_delivery.go

```diff
@@ -0,0 +1,131 @@
+type LogDeliveryConfiguration struct {
+	AccountID              string   `json:"account_id"`
+	ConfigID               string   `json:"config_id,omitempty" tf:"computed"`
+	CredentialsID          string   `json:"credentials_id"`
+	StorageConfigurationID string   `json:"storage_configuration_id"`
+	WorkspaceIdsFilter     []string `jsong:"workspace_ids_filter,omitempty"`
+	ConfigName             string   `json:"config_name,omitempty"`
+	Status                 string   `json:"status,omitempty"`
+	LogType                string   `json:"log_type"`
+	OutputFormat           string   `json:"output_format"`
+	DeliveryPathPrefix     string   `json:"delivery_path_prefix,omitempty"`
+	DeliveryStartTime      string   `json:"delivery_start_time,omitempty"`
+}
+
+// LogDeliveryAPI ...
+type LogDeliveryAPI struct {
+	client *common.DatabricksClient
+}
+
+// NewLogDeliveryAPI ...
+func NewLogDeliveryAPI(m interface{}) LogDeliveryAPI {
+	return LogDeliveryAPI{client: m.(*common.DatabricksClient)}
+}
+
+// Read reads log delivery configuration
+func (a LogDeliveryAPI) Read(accountID, configID string) (LogDeliveryConfiguration, error) {
+	var ld LogDelivery
+	err := a.client.Get(fmt.Sprintf("/accounts/%s/log-delivery/%s", accountID, configID), nil, &ld)
+	return ld.LogDeliveryConfiguration, err
+}
+
+// Create new log delivery configuration
+func (a LogDeliveryAPI) Create(ldc LogDeliveryConfiguration) (string, error) {
+	var ld LogDelivery
+	err := a.client.Post(fmt.Sprintf("/accounts/%s/log-delivery", ldc.AccountID), LogDelivery{
+		LogDeliveryConfiguration: ldc,
+	}, &ld)
+	// todo: verify with empty response - structs should have empty default strings
+	return ld.LogDeliveryConfiguration.ConfigID, err
+}
+
+// Disable log delivery configuration - e.g. delete it
+func (a LogDeliveryAPI) Disable(accountID, configID string) error {
+	return a.client.Patch(fmt.Sprintf("/accounts/%s/log-delivery/%s", accountID, configID), map[string]string{
+		"status": "DISABLED",
+	})
+}
+
+// ResourceLogDelivery ..
+func ResourceLogDelivery() *schema.Resource {
+	p := util.NewPairID("account_id", "config_id")
+	s := internal.StructToSchema(LogDeliveryConfiguration{},
+		func(s map[string]*schema.Schema) map[string]*schema.Schema {
+			s["account_id"].ForceNew = true
+			// nolint
+			s["config_name"].ValidateFunc = validation.StringLenBetween(0, 255)
+			return s
+		})
+	readContext := func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		accountID, configID, err := p.Unpack(d)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		ldc, err := NewLogDeliveryAPI(m).Read(accountID, configID)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		err = internal.StructToData(ldc, s, d)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		return nil
+	}
+	return &schema.Resource{
+		Schema:      s,
+		ReadContext: readContext,
+		CreateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```

Comment:
nit: why an inline func here but taking readContext out?

Reply:
because `readContext` is used twice (in read & create, usually it's also used in update). functions are inline here because they're all under 30 LoC & make no sense outside of resource definition.